packages/api/src/platforms/vtex/resolvers/validateSession.ts

```diff
@@ -7,6 +7,28 @@ import type {
   StoreSession,
 } from '../../../__generated__/schema'
 
+async function getGeoCoordinates(
+  clients: Context['clients'],
+  country: string,
+  postalCode: string
+) {
+  try {
+    const address = await clients.commerce.checkout.address({
+      postalCode,
+      country,
+    })
+
+    const [longitude, latitude] = address.geoCoordinates
+    return { latitude, longitude }
+  } catch (err) {
+    console.error(
+      'Error while getting geo coordinates for the current postal code and country.\n'
+    )
+
+    throw err
```

Comment:
question: If the call to Checkout's API fails, what do we want to happen?

Reply:
The GraphQL layer will handle it for us, but the request gets cancelled basically

Reply:
Great, it doesn't break the page then? And it would just stay with a null `geoCoordinates`?

Reply:
Yeah, but I need to double-check the return value indeed, I'll check and handle it if needed

Reply:
Validated that the FS GraphQL layer handle that and the request is cancelled.